src/main/java/org/springframework/data/elasticsearch/core/ResponseConverter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.elasticsearch.core;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.elasticsearch.cluster.metadata.MappingMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.springframework.data.elasticsearch.core.document.Document;
+import org.springframework.data.elasticsearch.core.mapping.IndexInformation;
+
+/**
+ * Factory class to elasticsearch responses to different type of data classes.
+ * @author George Popides
+ * @since 4.2
+ */
+public class ResponseConverter {
+	public ResponseConverter() {
+	}
+
+	// region alias
+
+	public List<IndexInformation> indexInformationCollection(GetIndexResponse getIndexResponse) {
+		List<IndexInformation> indexInformationList = new ArrayList<>();
+
+
+		for (String indexName : getIndexResponse.getIndices()) {
+			Document settings = settingsFromGetIndexResponse(getIndexResponse, indexName);
+			Document mappings = mappingsFromGetIndexResponse(getIndexResponse, indexName);
+			List<AliasMetadata> aliases = getIndexResponse.getAliases().get(indexName) != null
```

Comment:
Use `org.springframework.data.elasticsearch.core.index.AliasData` in these functions